compiler/vm/vm.nim

```diff
@@ -921,20 +931,21 @@ proc rawExecute(c: var TCtx, pc: var int, tos: var StackFrameIndex): YieldReason
     tos = nf
     updateRegsAlias
 
-  template guestValidate(cond: bool, msg: string) =
+  template guestValidate(cond: bool, pc: PrgCtr, strMsg: string) =
     ## Ensures that a guest-input related condition holds true and raises
     ## a `VmError` if it doesn't
     if unlikely(not cond): # Treat input violations as unlikely
       # TODO: don't use a string message here; use proper reports
-      raiseVmError(VMReport(kind: rvmErrInternal, str: msg))
+      raiseVmError(VmEvent(kind: vmEvtErrInternal, location: some pc, msg: strMsg))
```

Comment:
All this location setting should be wrapped up via a constructor.

Reply:
You don't need to. Events only leave the VM through `YieldReason`, which is constructed in `execute`. Setting the location can be done in a single place there.

As discussed on Matrix, I'd prefer not storing `location` as part of `VmEvent`. The call-site of `execute` can query the `TLineInfo` or program counter of the instruction the event happened as part of via either `source(c, thread)` or `thread.currentPc`, and then associate this data with the event however it wants.

In the case of VM execution inside the compiler (i.e. `compilerbridge`), the location can be queried in the `yrkError` branch in `compilerbridge.execute` and then passed to `buildError` where it is either stored directly inside the `ExecErrorReport` or used for the `VMReport`.